### PR TITLE
Add XRCompositor language to accomodate monoscopic displays

### LINKED ISSUE #68: How should layers behave on devices that have only 1 screen
For instance, projection layers are currently defined to always be `stereo` but that is not the case on mobile phones

### PR DESCRIPTION
**Pull request by cabanier**
+ clarified stereo 
+ fixed some typos

Closes #68

### CODE REVIEW

**Comment by Manishearth** [comment]
Hmm. This might be confusing, it might be worth erroring if these are used instead of mono?

Alternatively, we can perhaps rename these to not be "stereo" but instead be "views" where they each have the number of subrects equal to the number of views, and we can have `getViewport()`-esque stuff to get viewports instead of requiring right-left and top-bottom. It seems very weird to me that the core spec is super flexible on this but layers hardcodes the choices.

This would also properly solve https://github.com/immersive-web/layers/issues/114

**Comment by cabanier** [comment]
> Hmm. This might be confusing, it might be worth erroring if these are used instead of mono?

No, we don't want to error out because then mono experiences will crash unless the author specifically accommodates them.

> Alternatively, we can perhaps rename these to not be "stereo" but instead be "views" where they each have the number of subrects equal to the number of views, and we can have `getViewport()`-esque stuff to get viewports instead of requiring right-left and top-bottom. It seems very weird to me that the core spec is super flexible on this but layers hardcodes the choices.

I agree that `stereo` doesn't feel right but `views` is also confusing.
Maybe `Natural`, `all` or `default`?

**Comment by Manishearth** [comment]
yeah I don't quite care about the naming :)

It seems like we should just have "mono" vs "natural" and natural is laid out however the UA wants, using something like getViewport(). If we do this are the horizontal and vertical variants necessary?  I.e. does the user still need the ability to request a particular arrangement?

**Comment by cabanier** [comment]
> yeah I don't quite care about the naming :)
> 
> It seems like we should just have "mono" vs "natural" and natural is laid out however the UA wants, using something like getViewport().

Should I change it to `natural` or should we bring it to the group?

> If we do this are the horizontal and vertical variants necessary? I.e. does the user still need the ability to request a particular arrangement?

Yes, those are still needed to accommodate pre-canned video, bitmaps, etc

**Comment by cabanier** [comment]
/agenda What should we replace the `stereo` layout with?

**Comment by Manishearth** [comment]
> Yes, those are still needed to accommodate pre-canned video, bitmaps, etc

Should this not just use the XRMediaBinding stuff?

Even if we decide to have such layers, it still feels weird to then patch mono support in for these layers.

Overall these two seem to b pretty specal-case-y and I'm wary of adding something like that.

**Comment by cabanier** [comment]
> > Yes, those are still needed to accommodate pre-canned video, bitmaps, etc
> 
> Should this not just use the XRMediaBinding stuff?

No because authors will want to add other effects on top of video or use the stereo bitmaps.

> Even if we decide to have such layers, it still feels weird to then patch mono support in for these layers.
> 
> Overall these two seem to b pretty specal-case-y and I'm wary of adding something like that.

I'm not adding side-by-side layout in the PR. That was already discussed in previous issues.

**Comment by Manishearth** [comment]
> I'm not adding side-by-side layout in the PR. That was already discussed in previous issues.

Yes, but you are deciding a behavior for side-by-side layout textures in mono sessions, which is what's weird here.

And yeah I think this is a wider discussion that needs to be had: it was kinda ongoing in #114 but that was closed with a specific fix. I'll open something new.

**Comment by cabanier** [comment]
> > I'm not adding side-by-side layout in the PR. That was already discussed in previous issues.
> 
> Yes, but you are deciding a behavior for side-by-side layout textures in mono sessions, which is what's weird here.

If you have other behavior in mind, we can discuss that here.
Side-by-side layout should not break workflows that happen to be monoscopic.

**Comment by Manishearth** [comment]
>  Side-by-side layout should not break workflows that happen to be monoscopic.

Hmm. It still feels weird, but I won't block this. I'll file a separate issue for the existence of left-right and top-bottom.

**Comment by cabanier** [comment]
> > Side-by-side layout should not break workflows that happen to be monoscopic.
> 
> Hmm. It still feels weird, but I won't block this. I'll file a separate issue for the existence of left-right and top-bottom.

It was introduced by #46. We also discussed it in a weekly call but I don't know how to link that.

**Comment by Manishearth** [comment]
Yeah, I think there's a bit more to it than that. For example, even if we rename "stereo" there still needs to be a way to get viewports.

**Comment by Manishearth** [comment]
Hmm, looking at it it seems fine now, but we should probably add non-normative notes explaining these modes better.

**Comment by cabanier** [comment]
> Hmm, looking at it it seems fine now, but we should probably add non-normative notes explaining these modes better.

@Manishearth I added a note. Do you think more information is needed?

**Comment by Manishearth** [comment]
Looks great, thanks!

**Comment by cabanier** [comment]
We discussed this change a bit in this week's call but we didn't resolve what name we should use.
I'm leaning towards `default`

Also, @toji I was wrong when I said a projection layers can be mono on a stereo device. It's defined to be always `stereo` (aka `default`) so it has the same number of subimages as the session.

**Comment by cabanier** [comment]
@toji any comments on my latest changes?

File: webxrlayers-1.bs [hunk 1]
```diff
@@ -240,18 +240,22 @@ The {{XRLayerLayout}} enum defines the layout of the layer.
 <pre class="idl">
 enum XRLayerLayout {
   "mono",
-  "stereo",
+  "default",
```
Comment (toji) [inline webxrlayers-1.bs:243]: This feels fine to me, although as the most minor nit ever: If we're gonna call it "default" then we should probably make it the first item in the enum. :)

Comment (cabanier) [reply webxrlayers-1.bs:243]: ok :-)

**Pull request merged**